pkg/app/server/option.go

```diff
@@ -304,3 +305,19 @@ func WithDisablePrintRoute(b bool) config.Option {
 		o.DisablePrintRoute = b
 	}}
 }
+
+// WithOnAccept sets the callback function when a new connection is accepted but cannot
+// receive data in netpoll. In go net, it will be called before converting tls connection
+func WithOnAccept(fn func(conn network.Conn) context.Context) config.Option {
```

Comment:
This looks more like a Hook function and I wonder if it could be used in the same way as OnShutdownHook and OnRunHook.
such as https://github.com/cloudwego/hertz/blob/9b9ec92704e19626225f6fc10d69a62f1762dbda/pkg/route/engine.go#L185-L190

Reply:
Because it seems that Hook is now used in a different way if it is Hook function.

Reply:
No. Since netpoll and go net can bth divide into two phase. Please follow here. 


> In netpoll library, OnAccept is called after connection accepted but before adding it to epoll. OnConnect is called after adding it to epoll. The difference is that onConnect can get data but OnAccept cannot. If you'd like to check whether the peer IP is in the blacklist, you can use OnAccept. In go net, OnAccept is executed after connection accepted but before establishing tls connection. OnConnect is executed after establishing tls connection. 在 nepoll 中，OnAccept 会在连接 accept 后，加入 epoll 监听之前调用，这个时候可以获取到对端 ip 但是不能读取数据；onConnect 会在加入 epoll 监听之后使用。在 go net 当中，onAccept 会在连接 accept 之后，转换为 tls 连接之前（如果有的话）调用，onConnect 会在转换为 tls 连接之后调用